src/errors.ts

```diff
@@ -0,0 +1,13 @@
+export class SpOptionsEmptyError extends Error {
+    constructor(message?: string) {
+        super(message);
+        Error.captureStackTrace(this, SpOptionsEmptyError);
```

Comment:
To my knowledge `Error.captureStackTrace` is a non-standard v8 facility. If we want to support other js engines, we need to make this optional. Also is this needed? I don't think its needed if you are extending Error. `super(message)` should already be capturing the stack trace.

Reply:
Maybe we should test for the stack trace.

Reply:
Doing away with this